fmriprep/utils/misc.py

```diff
@@ -7,6 +7,29 @@
 """
 
 
+def remove_rotation_and_shear(img):
+    from transforms3d.affines import decompose, compose
+    import numpy as np
+
+    T, _, Z, _ = decompose(img.affine)
+    affine = compose(T=T, R=np.diag([1, 1, 1]), Z=Z)
+    return img.__class__(np.asanyarray(img.dataobj), affine, img.header)
+
+
+def split_and_rm_rotshear_func(in_file):
+    import os
+    import nibabel as nb
+    from fmriprep.utils.misc import remove_rotation_and_shear
+    out_files = []
+    imgs = nb.four_to_three(nb.load(in_file))
+    for i, img in enumerate(imgs):
+        out_file = os.path.abspath('vol%04d.nii.gz' % i)
+        img = remove_rotation_and_shear(img)
```

Comment:
I think we need to do `img = nb.as_closest_canonical(img)` before this. Consider an image with the affine:

```Python
>>> img.affine
array([[  0.        ,   3.        ,   0.        , -88.4322052 ],
       [ -1.66738188,   0.        ,   2.6871357 , -47.71268463],
       [  3.35891914,   0.        ,   1.33390486, -77.85404205],
       [  0.        ,   0.        ,   0.        ,   1.        ]])
>>> img.shape
(33, 64, 64)
```

```Python
>>> remove_rotation_and_shear(img).affine
array([[  3.75000002,   0.        ,   0.        , -88.4322052 ],
       [  0.        ,   3.        ,   0.        , -47.71268463],
       [  0.        ,   0.        ,   3.00000007, -77.85404205],
       [  0.        ,   0.        ,   0.        ,   1.        ]])

>>> remove_rotation_and_shear(nb.as_closest_canonical(img)).affine
array([[  3.        ,   0.        ,   0.        , -88.4322052 ],
       [  0.        ,   3.00000007,   0.        , -47.71268463],
       [  0.        ,   0.        ,   3.75000002, -77.85404205],
       [  0.        ,   0.        ,   0.        ,   1.        ]])
```

Notice that both are RAS matrices, but with different zooms. And if you look at the shapes:

```Python
>>> remove_rotation_and_shear(img).shape
(33, 64, 64)

>>> remove_rotation_and_shear(nb.as_closest_canonical(img)).shape
(64, 64, 33)
```

We clearly have a rotated data matrix, which cannot be true if both correctly have an RAS orientation. So we end up destroying all orientation information, not just the "off-diagonal" information, unless we rotate the data matrix first.

Reply:
Good catch